teachertool/src/services/loggingService.ts

```diff
@@ -0,0 +1,24 @@
+const formatMessageForConsole = (message: string) => {
+    const time = new Date();
+    return `[${time.getHours()}:${time.getMinutes()}:${time.getSeconds()}] ${message}`;
+}
+
+const formatName = (name: string) => {
+    return name.toLowerCase().replace(/ /g, "_");
+}
+
+export const logError = (name: string, details: string) => {
```

Comment:
Can we just log a message instead of name + message? I think logging to telemetry from here is too low level. Better to log errors where they happen.

Reply:
I actually think it's helpful to have a unified error log like this. My assumption is we'd call this when the error happens, and the "details" would be the relevant context from there (i.e. a stack trace, description, etc...). Maybe details should be a property bag instead of a string...but in essence I suppose I don't see this as "not logging errors when they happen" but rather, this could be the mechanism through which you log an error when it happens.

Reply:
But I think it's helpful to simplify queries around "What are the kinds of errors we're seeing in the teacher tool?" "Have there been any spikes in errors after a release?" and so on without having to query an every-changing set of different error events.